src/keys/get-management-token.ts

```diff
@@ -67,25 +70,45 @@ const getTokenFromOneTimeToken = async (
     `Successfully retrieved CMA Token for app ${appInstallationId} in space ${spaceId} and environment ${environmentId}`
   )
 
-  return JSON.parse(response.body).token
+  return JSON.parse(response.body).token as Promise<string>
 }
 
 /**
  * Factory method for GetManagementToken
  * @internal
  */
 export const createGetManagementToken = (log: Logger, http: HttpClient) => {
-  return (privateKey: unknown, opts: GetManagementTokenOptions): Promise<string> => {
+  return async (privateKey: unknown, opts: GetManagementTokenOptions): Promise<string> => {
     if (!(typeof privateKey === 'string')) {
       throw new ReferenceError('Invalid privateKey: expected a string representing a private key')
     }
 
+    const cacheKey = opts.appInstallationId + opts.environmentId + privateKey
+    if (opts.reuseToken !== false) {
+      const existing = cache.get(cacheKey) as string
+      if (existing !== undefined) {
+        return existing
+      }
+    }
```

Comment:
What about this?
```suggestion
      const existing = cache.get(cacheKey)
      if (existing) {
        return existing as string
      }
    }
```

Keeps the conditional clean and forces the type only if needed